sky/provision/gcp/instance_utils.py

```diff
@@ -114,6 +192,93 @@ def add_network_tag_if_not_exist(
     ) -> None:
         raise NotImplementedError
 
+    @classmethod
+    def create_instance(cls,
+                        cluster_name: str,
+                        project_id: str,
+                        availability_zone: str,
+                        node_config: dict,
+                        labels: dict,
+                        is_head_node: bool,
+                        wait_for_operation: bool = True) -> Tuple[dict, str]:
+        """Creates a single instance and returns result.
+
+        Returns a tuple of (result, node_name).
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def create_instances(
+        cls,
+        cluster_name: str,
+        project_id: str,
+        zone: str,
+        node_config: dict,
+        labels: dict,
+        count: int,
+        include_head_node: bool,
+        wait_for_operation: bool = True,
+    ) -> List[Tuple[dict, str]]:
+        """Creates multiple instances and returns result.
+
+        Returns a list of tuples of (result, node_name).
+        """
+        operations = [
+            cls.create_instance(cluster_name,
+                                project_id,
+                                zone,
+                                node_config,
+                                labels,
+                                is_head_node=include_head_node and i == 0,
+                                wait_for_operation=False) for i in range(count)
+        ]
```

Comment:
nit: add some loggings here, as it seems taking quite a long time when there are multiple instances to be provisioned.
```suggestion
        operations = []
        for i in range(count):
            logger.debug(f'Creating instance {i}/{count}')
            operations.append(cls.create_instance(cluster_name,
                                project_id,
                                zone,
                                node_config,
                                labels,
                                is_head_node=include_head_node and i == 0,
                                wait_for_operation=False)
```

Reply:
Or even using multiple threads:
```python
        from multiprocessing import pool
        with pool.ThreadPool() as p:
            operations = p.starmap(
            cls.create_instance,
            [
                (cluster_name, project_id, zone, node_config, labels,
                 include_head_node and i == 0, False)
                for i in range(count)
            ],
        )
```

It seems the provisioning time reduced from 200s to 90s by applying this multi-threading.

Reply:
resolved. same for another similar comment